spark-connect/client/src/test/scala-spark-master/io/delta/connect/tables/RemoteSparkSession.scala

```diff
@@ -97,10 +98,14 @@ trait RemoteSparkSession extends BeforeAndAfterAll { self: Suite =>
   override def beforeAll(): Unit = {
     super.beforeAll()
     server
-    // TODO: Instead of sleeping for a fixed time, which is a bit brittle,
-    // we should repeatedly check when the server is ready.
-    Thread.sleep(10000)
-    spark = SparkSession.builder().remote(s"sc://localhost:$serverPort").build()
+
+    val sparkConnectClient = SparkConnectClient
+      .builder()
+      .userId("test")
+      .port(serverPort)
+      .retryPolicy(RetryPolicy.defaultPolicy())
+
+    spark = SparkSession.builder().client(sparkConnectClient.build()).create()
```

Comment:
Why can't we just replace `.build()` with `.create()`? Why do we need to provide our own `SparkConnectClient`?

Reply:
@tomvanbussel 

https://github.com/apache/spark/blob/3edc9c23a723a92c5a951cea0436529de65c640a/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala#L1028

If we only do like this,

`SparkSession.builder().remote(s"sc://localhost:$serverPort").create()`

The only config it takes into account is `remote`

We need to include `client` to have `RetryPolicy` from `SparkConnectClient`

https://github.com/apache/spark/blob/3edc9c23a723a92c5a951cea0436529de65c640a/sql/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala#L722

`SparkSession`'s builder doesn't have `RetryPolicy`
https://github.com/apache/spark/blob/3edc9c23a723a92c5a951cea0436529de65c640a/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala#L887

Similar to here
https://github.com/apache/spark/blob/3edc9c23a723a92c5a951cea0436529de65c640a/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/test/RemoteSparkSession.scala#L181

Reply:
`remote` still uses the default retry policy, right?

Reply:
@tomvanbussel You are right, by default the `SparkSession`'s `builder` here already uses the default `Configuration` of the `SparkConnectClient` (https://github.com/apache/spark/blob/3edc9c23a723a92c5a951cea0436529de65c640a/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala#L891).

I thought that the old approach would not work because it was failing back then, with this error
`INVALID_HANDLE.SESSION_NOT_FOUND] The handle 746e6c86-9fa9-4b08-9572-388c20eaed47 is invalid. Session not found. SQLSTATE: HY000"`, but Spark patches this [error](https://github.com/apache/spark/pull/46971) a week later by still retrying even when encountering this exception, so our way back then should work even without the `Thread.sleep`